Automattic/pocket-casts-android
# Use debugProd only for application projects

## Description

A follow up to https://github.com/Automattic/pocket-casts-android/pull/2644#discussion_r1719145599. This PR removes `debugProd` variant from the library modules and configures it only for the apps. 

## Testing Instructions

Verify `debugProd` app variants. Check if mobile one still supports product info mocking and clip sharing.

## Checklist
 
#### I have tested any UI changes...
